app/server/api/vendor_payout_api.py

```diff
@@ -136,62 +154,69 @@ def post(self):
-                    'message': f'Invalid transfer amount!'
-                }
-                return make_response(jsonify(response_object)), 400
-
-            if line['Payment Has Been Made'].upper() == 'TRUE' and line['Bank Payment Date']:
-                try:
+                message = f'Transfer with ID {tid} not found!'
+                transfers.append((tid, None, message))
+                continue
+
+            if transfer.transfer_type != TransferTypeEnum.WITHDRAWAL:
+                message = f'Not a withdrawal!'
+                transfers.append((tid, None, message))
+                continue
+
+            got_amount = round(dollars_to_cents(line["UnitAmount"]))
+            expected_amount = round(transfer.transfer_amount)
+            if got_amount != expected_amount:
+                message = f'Transfer Amounts do not match (got {cents_to_dollars(got_amount)}, expected {cents_to_dollars(expected_amount)})!'
+                transfers.append((tid, None, message))
+                continue
+
+            try:
+                if line['Payment Has Been Made'].upper() == 'TRUE' and line['Bank Payment Date']:
                     transfer.resolve_as_complete_and_trigger_blockchain()
+                    message = 'Transfer Success'
+                elif line['Payment Has Been Made'] == 'FALSE':
+                    transfer.resolve_as_rejected()
+                    message = 'Transfer Rejected'
+            except Exception as e:
+                message = str(e)
 
-                except:
-                    pass
-            elif line['Payment Has Been Made'] == 'FALSE':
-                transfer.resolve_as_rejected()
-            transfers.append(transfer)
+            transfers.append((tid, transfer, message))
 
         output = io.StringIO()
         writer = csv.writer(output)
 
         writer.writerow([
-            'ID',
+            'Transfer ID',
+            'Vendor Account ID',
             'Phone',
             'First Name', 
             'Last Name', 
             'Transfer Created',
             'Transfer Type',
             'Transfer Amount',
             'Transfer Status',
+            'Message'
         ])
-        for t in transfers:
+        for tid, t, m in transfers:
             writer.writerow([
-                t.sender_transfer_account.id,
-                t.sender_transfer_account.primary_user.phone,
-                t.sender_transfer_account.primary_user.first_name,
-                t.sender_transfer_account.primary_user.last_name,
-                t.created,
-                t.transfer_type.value,
-                cents_to_dollars(t.transfer_amount),
-                t.transfer_status.value
+                tid,
+                t and t.sender_transfer_account.id,
+                t and t.sender_transfer_account.primary_user.phone,
+                t and t.sender_transfer_account.primary_user.first_name,
+                t and t.sender_transfer_account.primary_user.last_name,
```

Comment:
if t isn't defined does a `None` value cause an error?

Reply:
No it just leaves a blank space